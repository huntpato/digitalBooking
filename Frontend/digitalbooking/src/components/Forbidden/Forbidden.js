import React from 'react'
import { Link } from "react-router-dom";
import checked from '../../assets/forbidden.png'
import styles from  './Forbidden.module.css'

export const Forbidden = (props) => {
    const {confirmation, success, title, info, button} = styles;
    const {mes1, mes2, link, btn} = props;
    return (
        <div className= { confirmation }>
            <section className= { success }>
                <img src= { checked } alt='Ícono de reserva realizada.'/>
                <h3 className= { title }>{mes1}</h3>
                <h5 className= { info }>{mes2}</h5>
                <Link to= {link}><button className={ button }>{btn}</button></Link>
            </section>
        </div>
    )
}

Forbidden.defaultProps = {
    mes1: "Login necesario",
    mes2: "Para acceder a esta página se requiere iniciar sesión.",
    link: '/login',
    btn: "Iniciar sesión"
};