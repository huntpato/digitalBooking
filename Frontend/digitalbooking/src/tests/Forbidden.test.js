import React from "react";
import { shallow } from "enzyme";
import { Forbidden } from "../components/Forbidden/Forbidden";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";

describe("Pruebas en componente <Forbidden />", () => {
  const wrapper = shallow(<Forbidden />);

  test("Comprobar que <Forbidden /> renderiza correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });


});
