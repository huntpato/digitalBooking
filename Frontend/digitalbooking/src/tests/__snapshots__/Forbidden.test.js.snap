// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Forbidden /> Comprobar que <Forbidden /> renderiza correctamente. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="forbidden.png"
    />
    <h3
      className="title"
    >
      Login necesario
    </h3>
    <h5
      className="info"
    >
      Para acceder a esta página se requiere iniciar sesión.
    </h5>
    <Link
      to="/login"
    >
      <button
        className="button"
      >
        Iniciar sesión
      </button>
    </Link>
  </section>
</div>
`;
